test/mocks/http/mocks.h

```diff
@@ -388,6 +388,14 @@ class MockAsyncClientRequest : public AsyncClient::Request {
   MockAsyncClient* client_;
 };
 
+class MockAsyncClientStream : public AsyncClient::Stream {
+public:
```

Comment:
per gmock compile perf guidelines can we declare constructor/deconstructor here and then define in .cc file. For other mocks in this PR, I'm not sure if you can do that with template mocks, but if you can we should do that (probably not).